components/camel-drill/src/main/java/org/apache/camel/component/drill/DrillEndpoint.java

```diff
@@ -0,0 +1,177 @@
+ * limitations under the License.
+ */
+package org.apache.camel.component.drill;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Map;
+
+import oadd.org.apache.commons.lang.StringUtils;
+
+import org.apache.camel.Consumer;
+import org.apache.camel.Processor;
+import org.apache.camel.Producer;
+import org.apache.camel.component.drill.DrillComponent.DrillConnectionMode;
+import org.apache.camel.impl.DefaultPollingEndpoint;
+import org.apache.camel.spi.Metadata;
+import org.apache.camel.spi.UriEndpoint;
+import org.apache.camel.spi.UriParam;
+import org.apache.camel.spi.UriPath;
+import org.springframework.jdbc.core.ColumnMapRowMapper;
+import org.springframework.jdbc.core.RowMapperResultSetExtractor;
+
+
+/**
+ * The drill component gives you the ability to quering into apache drill
+ * cluster.
+ */
+@UriEndpoint(scheme = "drill", title = "DRILL", syntax = "drill:host", producerOnly = true, label = "database,sql")
+public class DrillEndpoint extends DefaultPollingEndpoint {
+
+    @UriPath(description = "Host name or IP address")
+    @Metadata(required = "true")
+    private String host;
+    @UriParam(description = "Port number")
+    @Metadata(required = "false", defaultValue = "2181")
+    private Integer port = 2181;
+    @UriParam(description = "Drill directory", defaultValue = "")
+    private String directory = "";
+    @UriParam(defaultValue = "")
+    private String clusterId = "";
+    @UriParam(defaultValue = "zk")
+    private String mode = "zk";
+
+    /**
+     * creates a drill endpoint
+     *
+     * @param uri the endpoint uri
+     * @param component the component
+     */
+    public DrillEndpoint(String uri, DrillComponent component) {
+        super(uri, component);
+    }
+
+    public Consumer createConsumer(Processor processor) throws Exception {
+        throw new UnsupportedOperationException("DrillConsumer is not supported!");
+    }
+
+    public Producer createProducer() throws Exception {
+        return new DrillProducer(this);
+    }
+
+    @Override
+    public boolean isSingleton() {
+        return true;
+    }
+
+    public String toJDBCUri() {
+        String url = "jdbc:drill:";
+        if (mode.toUpperCase().equals(DrillConnectionMode.DRILLBIT.name())) {
+            // TODO JIRA BUG connection mode
+            url += mode + "=" + host;
+        } else {
+            url += mode + "=" + host + ":" + port;
+        }
+
+        if (StringUtils.isNotBlank(directory)) {
+            url += "/" + directory;
+        }
+        if (StringUtils.isNotBlank(clusterId)) {
+            url += "/" + clusterId;
+        }
+
+        return url;
+    }
+
+    @SuppressWarnings("unchecked")
+    public List<?> queryForList(ResultSet rs) throws SQLException {
+        ColumnMapRowMapper rowMapper = new ColumnMapRowMapper();
+        RowMapperResultSetExtractor<Map<String, Object>> mapper = new RowMapperResultSetExtractor<Map<String, Object>>(rowMapper);
```

Comment:
@lburgazzoli this code is an extract of SqlProducer.java code.

Yes, it is also nice! After this PR i could create another JIRA to improve component.